stories/demos/CommonFeatures.tsx

```diff
@@ -253,29 +250,42 @@ export function CommonFeatures() {
       case 'transaction':
       case 'account':
       case 'version':
-        sortedRows = sortedRows.sort((a, b) => a[sortColumn].localeCompare(b[sortColumn]));
-        break;
+        return function (a: Row, b: Row) {
+          return a[sortColumn].localeCompare(b[sortColumn]);
+        };
       case 'available':
-        sortedRows = sortedRows.sort((a, b) =>
-          a[sortColumn] === b[sortColumn] ? 0 : a[sortColumn] ? 1 : -1
-        );
-        break;
+        return function (a: Row, b: Row) {
+          return a[sortColumn] === b[sortColumn] ? 0 : a[sortColumn] ? 1 : -1;
+        };
       case 'id':
       case 'progress':
       case 'startTimestamp':
       case 'endTimestamp':
       case 'budget':
-        sortedRows = sortedRows.sort((a, b) => a[sortColumn] - b[sortColumn]);
-        break;
+        return function (a: Row, b: Row) {
+          return a[sortColumn] - b[sortColumn];
+        };
       default:
+        throw new Error(`unsupported sortColumn: "${sortColumn}"`);
     }
+  };
 
-    return sortDirection === 'DESC' ? sortedRows.reverse() : sortedRows;
-  }, [rows, sortDirection, sortColumn]);
+  const sortedRows: readonly Row[] = useMemo(() => {
```

Comment:
Nitpicky, but moving the type closer to where a type issue may occur will give us better type errors.
```suggestion
  const sortedRows = useMemo((): readonly Row[] => {
```